packages/go_router/test/builder_test.dart

```diff
@@ -272,6 +272,54 @@ void main() {
       expect(find.byType(_HomeScreen), findsNothing);
       expect(find.byType(_DetailsScreen), findsOneWidget);
     });
+
+    testWidgets('Builds a HeroControllerScope for ShellRoute',
```

Comment:
Can you also add a check that hero will be trigger if under shell route?

Reply:
@chunhtai I'm sorry holidays are here at the moment, if you could give me a suggestion how to do it, I will commit that.

Reply:
You can have two sub goRoutes under a shellroute

lets say GoRouteA and GoRouteB, makes sure both GoRoute Build a Hero widget with the same tag and with flightShuttleBuilder to build some thing.

When they you do a `go` from GoRouteA to GoRouteB. During the page transition, you can check whether the widget that built in flightShuttleBuilder can be found through Find.

Reply:
I tried to write a test based on your comment @chunhtai, but it keeps failing. I am unsure if the pumping step works correctly, or if the `router.go` navigation completes before I can check if the key is present.  Currently it keeps failing, because it does not find the heroKey after calling `.go` and `.pump`

 ```
testWidgets('Triggers a Hero inside a ShellRoute',
        (WidgetTester tester) async {
      final UniqueKey heroKey = UniqueKey();

      final List<RouteBase> routes = <RouteBase>[
        ShellRoute(
          builder: (BuildContext context, GoRouterState state, Widget child) {
            return child;
          },
          routes: <GoRoute>[
            GoRoute(
                path: '/a',
                builder: (BuildContext context, _) {
                  return Hero(
                    tag: 'hero',
                    child: Container(),
                    flightShuttleBuilder: (_, __, ___, ____, _____) {
                      return Container(key: heroKey);
                    },
                  );
                }),
            GoRoute(
                path: '/b',
                builder: (BuildContext context, _) {
                  return Hero(
                    tag: 'hero',
                    child: Container(),
                  );
                }),
          ],
        )
      ];
      final GoRouter router =
          await createRouter(routes, tester, initialLocation: '/a');
      await tester.pumpAndSettle();

      // check that flightShuttleBuilder widget is not yet present
      expect(find.byKey(heroKey), findsNothing);

      // start navigation
      await tester.pump();
      router.go('/b');
      await tester.pump(const Duration(milliseconds: 30));
      expect(find.byKey(heroKey), isOnstage);

      // Waits for the animation finishes.
      await tester.pumpAndSettle();
      expect(find.byKey(heroKey), findsNothing);
    });
```

Reply:
@chunhtai @angjelkom I have fixed the test (something about when to `pump` was the issue. Can someone validate, commit and then merge this PR? 

```dart
testWidgets('Triggers a Hero inside a ShellRoute',
        (WidgetTester tester) async {
      final UniqueKey heroKey = UniqueKey();
      const String kHeroTag = 'hero';

      final List<RouteBase> routes = <RouteBase>[
        ShellRoute(
          builder: (BuildContext context, GoRouterState state, Widget child) {
            return child;
          },
          routes: <GoRoute>[
            GoRoute(
                path: '/a',
                builder: (BuildContext context, _) {
                  return Hero(
                    tag: kHeroTag,
                    child: Container(),
                    flightShuttleBuilder: (_, __, ___, ____, _____) {
                      return Container(key: heroKey);
                    },
                  );
                }),
            GoRoute(
                path: '/b',
                builder: (BuildContext context, _) {
                  return Hero(
                    tag: kHeroTag,
                    child: Container(),
                  );
                }),
          ],
        )
      ];
      final GoRouter router =
          await createRouter(routes, tester, initialLocation: '/a');

      // check that flightShuttleBuilder widget is not yet present
      expect(find.byKey(heroKey), findsNothing);

      // start navigation
      router.go('/b');
      await tester.pump();
      await tester.pump(const Duration(milliseconds: 10));
      // check that flightShuttleBuilder widget is visible
      expect(find.byKey(heroKey), isOnstage);
      // // Waits for the animation finishes.
      await tester.pumpAndSettle();
      expect(find.byKey(heroKey), findsNothing);
    });
```